frontend/src/Components/LibraryCard.tsx

```diff
@@ -40,12 +43,40 @@ export default function LibraryCard({
         }
     };
     const openContentProviderName =
-        library?.open_content_provider.name.charAt(0).toUpperCase() +
-        library?.open_content_provider.name.slice(1);
+        library?.open_content_provider_name.charAt(0).toUpperCase() +
+        library?.open_content_provider_name.slice(1);
+
+    function toggleFavorite(e: MouseEvent) {
+        e.preventDefault();
+        API.put(`open-content/${library.id}/save`, {
+            name: library.name,
+            content_id: library.id,
+            open_content_provider_id: library.open_content_provider_id,
+            content_url: `/api/proxy/libraries/${library.id}/`
+        })
+            .then(() => {
+                setFavorited(!favorited);
+                library.is_favorited = !favorited;
+            })
+            .catch((error) => console.error(error));
+        void mutate();
+    }
+
+    const favoriteIcon = library.is_favorited ? (
+        <StarIcon
+            className="w-5 text-primary-yellow cursor-pointer"
+            onClick={toggleFavorite}
+        />
+    ) : (
+        <StarIconOutline
+            className="w-5 cursor-pointer"
+            onClick={toggleFavorite}
+        />
+    );
 
     return (
         <div
-            className="card overflow-hidden cursor-pointer"
```

Comment:
keep `overflow-hidden` in here, unless there is a specific reason you deleted it I'm not aware of.

Reply:
I deleted it because the tool-tip was being rendered behind, so you only saw a very small piece of the tool-tip.

Reply:
oohhhh okay good point! okay to leave it as is then!